### PR TITLE
Add type annotations and static type checking

### LINKED ISSUE #23: Add type annotations to data model
Once our data model starts stabilizing a bit more, we should add type annotations from the [`typing`](https://docs.python.org/3/library/typing.html) standard library module and add type checking to our Travis builds.

### PR DESCRIPTION
**Pull request by cheukyin699**
# Pull Request

## Description

Add type annotations (that are sometimes somewhat complicated with the `Type[User]` and `Union` and whatnot). They are checked using `mypy` python static type checker.

Feel free to question me.

## Testing

After updating your pipenv, you can run `scripts/build_check.sh`. They should run correctly.

## Ticket(s)

Closes #23.

### CODE REVIEW

File: tests/command/commands/user_test.py [hunk 1]
```diff
@@ -194,26 +194,6 @@ def test_handle_edit_github(self):
         self.mock_facade.store.assert_called_once_with(user)
         self.mock_github.org_add_member.assert_called_once_with("rob")
 
-    def test_handle_edit_github_error(self):
```
Comment (rwblickhan) [inline tests/command/commands/user_test.py]: Why was this removed?

File: command/commands/token.py [hunk 2]
```diff
@@ -18,7 +21,9 @@ class TokenCommand:
     success_msg = "This is your token:\n```\n{}\n```" \
                   "\nKeep it secret! Keep it safe!\nIt will expire at {}."
 
-    def __init__(self, db_facade, config):
+    def __init__(self,
+                 db_facade: DBFacade,
+                 config: 'TokenCommandConfig') -> None:
```
[the remark this answers is not in the document]
Comment (cheukyin699) [reply command/commands/token.py]: Yes. It's for forward references. For some reason, just running mypy with some default settings allows you to take away the quotes (because it's referencing its own class). But when you run pytest with mypy plugin, it doesn't recognize its own class. Weird.

File: tests/command/commands/user_test.py (hunk 1 above)
Comment (cheukyin699) [reply tests/command/commands/user_test.py]: Because it is a duplicate function (overridden immediately by the one directly below it). It seems to be a remnant of some merge conflict (woooo spooky).

File: command/core.py [hunk 3]
```diff
@@ -1,26 +1,37 @@
...
 
 class Core:
     """Encapsulate methods for handling events."""
 
-    def __init__(self, db_facade, bot, gh_interface, token_config):
+    def __init__(self,
+                 db_facade: DBFacade,
+                 bot: Bot,
+                 gh_interface: GithubInterface,
+                 token_config) -> None:
```
[the remark this answers is not in the document]
Comment (cheukyin699) [reply command/core.py]: Yes. Let me check.

File: tests/command/commands/user_test.py (hunk 1 above)
Comment (rwblickhan) [reply tests/command/commands/user_test.py]: Oh haha didn't notice that, nice catch.

File: command/commands/team.py [hunk 4]
```diff
@@ -173,7 +181,8 @@ def handle(self, command, user_id):
         else:
             return self.get_help(), 200
 
-    def create_helper(self, param_list):
+    def create_helper(self,
+                      param_list: Dict[str, str]) -> ResponseTuple:
         """
```
Comment (jacqueschen1) [inline command/commands/team.py]: Some of the changes here may conflict with PR for finishing team commands, not sure if you wanna push these now and we could merge later or to remove the changes for now.

File: interface/slack.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 class Bot:
     """Utility class for calling Slack APIs."""
 
-    def __init__(self, sc: SlackClient) -> None:
+    def __init__(self, sc: SlackClient, slack_channel: str ="") -> None:
```
Comment (ravenCondol) [inline interface/slack.py]: fails pycodestyle  !

File: config.toml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-testing = false
```
Comment (rwblickhan) [inline config.toml]: I don't think we should make this change - it'll probably clobber the config that's present on the server.

Comment (rwblickhan) [reply config.toml]: More generally, though, I think we should probably figure out a better way to do this integration testing - we could talk about that next week though.

File: command/commands/token.py [hunk 7]
```diff
@@ -30,11 +35,13 @@ def __init__(self, db_facade, config):
...
 
-    def handle(self, _command, user_id):
+    def handle(self,
+               _command: str,
+               user_id: str) -> ResponseTuple:
         """Handle request for token."""
         logging.debug("Handling token command")
         try:
-            user = self.facade.retrieve(User, user_id)
+            user = cast(User, self.facade.retrieve(User, user_id))
```
Comment (rmcreyes) [inline command/commands/token.py]: I don't know much about how typing works, but would it be possible for the facade to cast the object it retrieves from the database before returning it? Saves from having to cast in every instance the facade returns something.

**Comment by ravenCondol** [comment]
WHAT DID I JUST DO
